scripts/limonade/src/index.ts

```diff
@@ -0,0 +1,130 @@
+    description: "Global config of limo program to filter for.",
+    type: "string",
+    demandOption: true,
+  })
+  .option("chain-id", {
+    description: "Chain id to publish these orders for as opportunity",
+    type: "string",
+    demandOption: true,
+  })
+  .option("api-key", {
+    description:
+      "API key to authenticate with the express relay server for publishing opportunities.",
+    type: "string",
+    demandOption: true,
+  })
+  .help()
+  .alias("help", "h")
+  .parseSync();
+
+async function run() {
+  const connection = new Connection(argv.rpcEndpoint);
+
+  const globalConfig = new PublicKey(argv.globalConfig);
+  let filters: GetProgramAccountsFilter[] = [
+    {
+      memcmp: {
+        bytes: globalConfig.toBase58(),
+        offset: 8,
+      },
+    },
+    {
+      memcmp: {
+        bytes: bs58.encode(Order.discriminator),
+        offset: 0,
+      },
+    },
+  ];
+
+  console.log("Listening for program account changes");
+  const client = new Client({ baseUrl: argv.endpoint, apiKey: argv.apiKey });
+  let { blockhash: latestBlockhash } = await connection.getLatestBlockhash(
+    "confirmed"
+  );
+  const submitExistingOpportunities = async () => {
+    const response = await connection.getProgramAccounts(limoId, {
+      commitment: "confirmed",
+      filters,
+      withContext: true,
+    });
+    const payloads: OpportunityCreate[] = response.value
+      .map((account) => ({
+        program: "limo" as const,
+        blockHash: latestBlockhash,
+        chainId: argv.chainId,
+        slot: response.context.slot,
+        order: {
+          state: Order.decode(account.account.data),
+          address: account.pubkey,
+        },
+      }))
+      .filter(
+        (opportunityCreate) =>
+          opportunityCreate.order.state.remainingInputAmount.toNumber() !== 0
+      );
+
+    console.log("Initial opportunities", payloads.length);
+    for (const payload of payloads) {
+      await client.submitOpportunity(payload);
+    }
+  };
+
+  connection.onProgramAccountChange(
+    limoId,
+    async (info, context) => {
+      console.log("Program account changed", info);
+      const order = Order.decode(info.accountInfo.data);
+      if (order.remainingInputAmount.toNumber() === 0) {
+        return;
+      }
+      console.log(order);
+
+      const payload: OpportunityCreate = {
+        program: "limo",
+        blockHash: latestBlockhash,
+        chainId: argv.chainId,
+        slot: context.slot,
+        order: { state: order, address: info.accountId },
+      };
+
+      await client.submitOpportunity(payload);
+    },
+    "confirmed",
+    filters
+  );
+  const updateLatestBlockhash = async () => {
+    while (true) {
+      latestBlockhash = (await connection.getLatestBlockhash("confirmed"))
+        .blockhash;
+      await new Promise((resolve) => setTimeout(resolve, 10000));
+    }
+  };
+  await Promise.all([updateLatestBlockhash(), submitExistingOpportunities()]);
```

Comment:
You can use the then, catch pattern here for the `submitExistingOpportunities` as well.